ledger/src/blockstore_db.rs

```diff
@@ -2273,32 +2282,16 @@ mod rocks_metrics_utils {
         },
         std::cell::RefCell,
     };
-    const METRIC_SAMPLES_1K: i32 = 1000;
-    // The default number of rocksdb perf samples in 1K
-    const ROCKSDB_PERF_CONTEXT_SAMPLES_IN_1K_DEFAULT: i32 = 0;
-    lazy_static! {
-    // The number of RocksDB performance counter samples in 1000.
-    static ref ROCKSDB_PERF_CONTEXT_SAMPLES_IN_1K: i32 =
-    std::env::var("SOLANA_METRICS_ROCKSDB_PERF_SAMPLES_IN_1K")
-        .map(|x| {
-            x.parse().expect("Failed to parse SOLANA_METRICS_ROCKSDB_PERF_SAMPLES_IN_1K")
-
-        }).unwrap_or(ROCKSDB_PERF_CONTEXT_SAMPLES_IN_1K_DEFAULT);
-
-    }
 
     // Thread local instance of RocksDB's PerfContext.
     thread_local! {static PER_THREAD_ROCKS_PERF_CONTEXT: RefCell<PerfContext> = RefCell::new(PerfContext::default());}
 
-    /// The function enables RocksDB's PerfContext in N out of 1000
-    /// where N is ROCKSDB_PERF_CONTEXT_SAMPLES_IN_1K.
-    ///
     /// Returns true if the PerfContext is enabled.
-    pub fn maybe_collect_perf_context() -> bool {
-        if *ROCKSDB_PERF_CONTEXT_SAMPLES_IN_1K <= 0 {
+    pub(crate) fn maybe_collect_perf_context(sample_interval: &u32) -> bool {
```

Comment:
`pub(crate)` is not needed. this function should be private to this module.
`u32` is small and implements `Copy`. so `sample_interval: u32`, not `&u32`.

Reply:
This function is inside a different mod, which we will later move to a different file where we have all the metric related stuff.

Reply:
> u32 is small and implements Copy. so sample_interval: u32, not &u32.

Aghh, yep.  I forgot the fact that it implements Copy.  Using passing and avoiding the `*` operator is definitely better.